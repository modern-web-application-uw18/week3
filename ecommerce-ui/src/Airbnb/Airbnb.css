.myCart {
  position: fixed;
  height: 80%;
  width: 80%;
  /* top: 3%; */
  /* left: 6%; */
  background: white;
  border: 5px dotted rgb(128, 128, 128);
  /* text-align: center; */
  border-radius: 20px;
}

.modal {
  background: rgba(0, 0, 0, .75);
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  visibility: hidden;
  opacity: 0;
  transition: visibility 0s linear 0.33s, opacity 0.33s linear;
  display: flex;
  justify-content: center;
  align-items: center;
}

.hide {
  visibility: hidden;
  opacity: 0;
}

.show {
  visibility: visible;
  opacity: 1;
  transition-delay: 0s;
}

.myCartContents {
  margin: 0 auto;
  width: 100%;
  height: 90%;
  display: flex;
  flex-flow: row;
  flex-wrap: wrap;
  justify-content: space-around;
  overflow: auto;
}

.myCartContents h2 {
  width: 100%;
}

.myCartContents .card {
  height: 380px;
}

.closeCart {
  position: absolute;
  top: -20px;
  right: -20px;
  font-size: 2rem;
  cursor: pointer;
}

.closeCart .fa-times-circle {
  color: red;
}

.closeCart .fa-circle {
  color: white;
}
