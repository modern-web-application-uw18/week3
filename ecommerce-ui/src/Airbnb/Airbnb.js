import React, { Component } from 'react';

import airbnbs from '../resources/airbnbs.json';
import articles from '../resources/articles.json';
import quickSearch from '../resources/quickSearch.json';
import navigation from '../resources/navigation.json';

import Navigation from '../Navigation/Navigation.js';
import Card from '../Card/Card.js';

import SectionBrowse from '../SectionBrowse/SectionBrowse.js';
import SectionArticles from '../SectionArticles/SectionArticles.js';
import SectionQuickSearch from '../SectionQuickSearch/SectionQuickSearch.js';

import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faTimesCircle from '@fortawesome/fontawesome-free-solid/faTimesCircle';
import faCircle from '@fortawesome/fontawesome-free-solid/faCircle';

import './Airbnb.css';

import {
  BrowserRouter as Router, Route
} from 'react-router-dom';

export default class Airbnb extends Component {
    constructor(props){
    	super(props);
    	this.state = {
        myCartItems: [],
        noItems: 0
      };
    }

    /**
    * adds a new object to the myCartItems array
    * @param {object} obj - the object to save into the shopping cart
    * @return {object} the updated shopping cart with the new obj added.
    */
    addToCart = (obj) => {
      return () => {
          this.setState((prevState, props) => {
            let newCart = prevState.myCartItems; // save previous state AS newCart variable
            newCart.push(obj); // add new item to old shopping cart which makes it a NewCart =]
            return {  // return the new cart to save the state
              myCartItems: newCart,
              noItems: newCart.length
            }
          });
      }
    }

    showCart = () => {
      const cartModal = document.getElementById('myShoppingCart');
      cartModal.classList.remove('hide');
      cartModal.classList.add('show');
    }

    hideCart = () => {
      const cartModal = document.getElementById('myShoppingCart');
      cartModal.classList.remove('show');
      cartModal.classList.add('hide');
    }

    render() {
        const myCart = this.state.myCartItems;
        const myItems = myCart.map((place, index) => {
          return <Card place={place} image={place.image} key={index} index={index}/>
        });

        return (
          <div>
            <nav>
              <Navigation navItems={navigation} noItems={this.state.noItems} showCart={this.showCart}/>
            </nav>
            <main>
              <Router>
                <div>
                  <Route path="/articles" render={() => <SectionArticles articles={articles} />} />
                  <Route path="/searches" render={() => <SectionQuickSearch quickSearch={quickSearch} />} />
                  <Route path="/" render={() => <SectionBrowse airbnbs={airbnbs} addToCart={this.addToCart} />} />
                </div>
              </Router>


              <div id="myShoppingCart" className="modal hide">
                <div className="myCart">
                  <h2>Your Shopping Cart</h2>
                  <div className="myCartContents">
                    {myItems}
                    <div className="closeCart" onClick={this.hideCart}>
                      <span className="fa-layers fa-fw">
                        <FontAwesomeIcon icon={faCircle} />
                        <FontAwesomeIcon icon={faTimesCircle} />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </main>
          </div>
        );
    }
}
