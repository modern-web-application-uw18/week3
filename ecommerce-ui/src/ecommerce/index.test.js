import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Airbnb from './index'

describe("Airbnb", () => {
	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<Airbnb />, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	it ('Should render with correct info on the individual properties', () => {
		const component = TestRenderer.create(<Airbnb />);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
