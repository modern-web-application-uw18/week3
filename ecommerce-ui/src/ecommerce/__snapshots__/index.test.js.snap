// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airbnb Should render with correct info on the individual properties 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-name"
    >
      Airbnb(clone)
    </h1>
  </header>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <section>
        <form>
          <div>
            <h2>
              Properties in your cart:
            </h2>
            <div>
              <h1>
                Cart total: 
                $0
              </h1>
            </div>
          </div>
        </form>
      </section>
      <div
        className="col-4"
      >
        <div
          className="list"
        >
          <div
            className="row"
          >
            <div
              className="col-4 items"
            >
              <h2>
                Centrally-located Manhattan studio
              </h2>
              <img
                alt="Centrally-located Manhattan studio"
                src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
              />
              <p>
                Host: 
                Robert
                , offering their: 
                Apartment
              </p>
              <p>
                Where: 
                New York
                , 
                USA
              </p>
              <h4>
                Cost: 
                $105
              </h4>
              <button
                className="cartBtn"
                onClick={[Function]}
                value={0}
              >
                Add to cart!
              </button>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-4 items"
            >
              <h2>
                Denver apartment outside Union Station
              </h2>
              <img
                alt="Denver apartment outside Union Station"
                src="https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80"
              />
              <p>
                Host: 
                Jamie
                , offering their: 
                Apartment
              </p>
              <p>
                Where: 
                Denver
                , 
                USA
              </p>
              <h4>
                Cost: 
                $130
              </h4>
              <button
                className="cartBtn"
                onClick={[Function]}
                value={1}
              >
                Add to cart!
              </button>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-4 items"
            >
              <h2>
                Quaint 3-bedroom home in Gamla stan
              </h2>
              <img
                alt="Quaint 3-bedroom home in Gamla stan"
                src="https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80"
              />
              <p>
                Host: 
                Olivia
                , offering their: 
                Entire House
              </p>
              <p>
                Where: 
                Stockholm
                , 
                Sweden
              </p>
              <h4>
                Cost: 
                $230
              </h4>
              <button
                className="cartBtn"
                onClick={[Function]}
                value={2}
              >
                Add to cart!
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
